### ISSUE TITLE
Support required arguments with default values

### ISSUE THREAD

**Issue by bessey** [rmosolgo/graphql-ruby#2356]
The GraphQL spec has [for some time supported required arguments with default values](https://github.com/graphql/graphql-js/pull/1274), however graphql-ruby (1.9) appears not to support this

```ruby
argument :id, ID, required: true, default_value: "123"
```
results in the schema compilation error:
`Query is invalid: field "account" argument "id" Variable id of type "ID!" is required and will not use the default value. Perhaps you meant to use type "ID".`

This seems like a very legitimate use case to me, a less contrived example:
```ruby
argument :hideDisabledAccounts, Boolean, required: true, default_value: false
# vs
argument :hideDisabledAccounts, Boolean, required: false
```
As Rubyists we understand that nil is false-y, so the behaviour here is roughly the same, but from the outside (and the inside) the 1st definition's behaviour seems to me much clearer than the 2nd, in the case where no value is provided.

References:
[This behaviour being requested in GraphQL reference implementation, and delivered](https://github.com/graphql/graphql-js/issues/1405)
[Merged spec change which adds "Non-null arguments and input object fields are no longer required to be provided a value/variable if that argument also supplies a default value."](https://github.com/graphql/graphql-spec/pull/418)

**Comment by rmosolgo** [comment]
Thanks for opening this conversation and including those links. It looks like I missed that spec update 🙈 I agree it should work the way the spec says!

**Comment by modosc** [comment]
@rmosolgo any chance you could check out my pr? we ran into some issues recently that this would help solve, thanks!

**Comment by modosc** [comment]
@rmosolgo are there any blockers from getting this released that i can help out with?

**Comment by rmosolgo** [comment]
No blockers, just released 1.9.10 🚢 !

**Comment by modosc** [comment]
thanks! 💃
